### PR TITLE
in-process hooks with injector support

### PR DESCRIPTION
**Pull request by tailsu**
* hooks that have a 'module.exports = ...' statement are executed in-process and the exported function is injected
* global Promise polyfill
* hooked method arguments are injected into hooks through the 'hookArgs' argument

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "cli-table": "https://github.com/telerik/cli-table/tarball/v0.3.1.1",
     "colors": "0.6.2",
     "cookie": "0.1.0",
+    "esprima": "^2.7.0",
```
Comment (teobugslayer) [inline package.json]: Please use exact version

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by rosen-vladimirov** [comment]
:+1:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Pull request merged**